CRM/Contribute/BAO/Contribution.php

```diff
@@ -1105,21 +1105,19 @@ private static function updateFinancialAccountsOnContributionStatusChange(&$para
     $previousContributionStatus = CRM_Contribute_PseudoConstant::contributionStatus($params['prevContribution']->contribution_status_id, 'name');
     $currentContributionStatus = CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $params['contribution']->contribution_status_id);
 
-    $isARefund = FALSE;
     if ((($previousContributionStatus == 'Partially paid' && $currentContributionStatus == 'Completed')
         || ($previousContributionStatus == 'Pending refund' && $currentContributionStatus == 'Completed')
         // This concept of pay_later as different to any other sort of pending is deprecated & it's unclear
         // why it is here or where it is handled instead.
         || ($previousContributionStatus == 'Pending' && $params['prevContribution']->is_pay_later == TRUE
           && $currentContributionStatus == 'Partially paid'))
     ) {
-      return [FALSE, $isARefund];
+      return [FALSE, FALSE];
     }
 
-    if ($previousContributionStatus == 'Completed'
-      && (self::isContributionStatusNegative($params['contribution']->contribution_status_id))
-    ) {
-      $isARefund = TRUE;
+    $isARefund = $previousContributionStatus === 'Completed' && self::isContributionStatusNegative($params['contribution']->contribution_status_id);
```

Comment:
do we need to wrap this in a containing set of `()` @eileenmcnaughton ?

Reply:
I don't think so - it's and AND

Reply:
@mattwire the brackets go around everything on the the right side of '=' = phpstorm will be noisy about that - I can extract to a funciton to avoid that